src/task-actions.ts

```diff
@@ -28,6 +28,7 @@ import {
 } from "hardhat/types";
 import { getWalletUtil } from "./extend-utils";
 import { createIntegratedDevnet } from "./devnet";
+import { handleCache, saveCacheEntry } from "./utils/recompiler";
```

Comment:
Only now have I realized that `recompiler.ts` is located within a `utils` directory. Now we have a `utils` file and a `utils` directory. I would suggest having `recompiler.ts` directly in `src`.